src/app/container/product-list/product-list.component.ts

```diff
@@ -0,0 +1,550 @@
+import { Component, Input } from '@angular/core';
+import { Product } from 'src/app/Models/Product';
+
+@Component({
+  selector: 'app-product-list',
+  templateUrl: './product-list.component.html',
+  styleUrls: ['./product-list.component.css']
+})
+export class ProductListComponent {
+  selectedProduct: Product
+  products = [
```

Comment:
It is very bad that you are storing this data in component, at least move it to another file